test/e2e-library.sh

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eo pipefail
+
+source "$(dirname $0)"/../hack/test-env.sh
+
+# TODO(@carlisia): refactor these tests
+
+function e2e_istio() {
+    export GATEWAY_OVERRIDE=""
+    export GATEWAY_NAMESPACE_OVERRIDE=""
+    export CONTROL_NAMESPACE=knative-serving
+
+    failed=0
+    go_test_e2e -timeout=20m -tags=e2e -parallel=12 \
+        ./test/conformance \
+        --enable-alpha --enable-beta \
+        --skip-tests="${ISTIO_UNSUPPORTED_E2E_TESTS}" \
+        --ingressClass=gateway-api.ingress.networking.knative.dev || failed=1
+
+    # Give the controller time to sync with the rest of the system components.
+    sleep 30
+
+    go_test_e2e -timeout=15m -failfast -parallel=1 ./test/ha -spoofinterval="10ms" \
+    --ingressClass=gateway-api.ingress.networking.knative.dev || failed=1
+
+    (( failed )) && dump_cluster_state
+    (( failed )) && fail_test
+
+    success
+}
+
+# The tests below are run in a Kind cluster, therefore they need to have their ingressendpoint and cluster-suffix specified.
+function cluster_info() {
+    CONTROL_NAMESPACE=knative-serving
+    IPS=( $(kubectl get nodes -lkubernetes.io/hostname!=kind-control-plane -ojsonpath='{.items[*].status.addresses[?(@.type=="InternalIP")].address}') )
+    CLUSTER_SUFFIX=${CLUSTER_SUFFIX:-cluster.local}
+}
```

Comment:
`cluster_info` here feels very similar to `export_cluster_info` in `e2e-common.sh`. Combine?

Reply:
(Feel free to make this a TODO if it feels too fragile to do right now)